tests/ui/unintentional_unit_return.rs

```diff
@@ -0,0 +1,21 @@
+#![warn(clippy::unintentional_unit_return)]
+#![feature(is_sorted)]
+
+struct Struct {
+    field: isize,
+}
+
+fn double(i: isize) -> isize {
+    i * 2
+}
+
+fn main() {
+    let mut structs = vec![Struct { field: 2 }, Struct { field: 1 }];
+    structs.sort_by_key(|s| {
+        double(s.field);
+    });
+    structs.sort_by_key(|s| double(s.field));
+    structs.is_sorted_by_key(|s| {
+        double(s.field);
+    });
```

Comment:
Can you add a check for a return statement in the closure:
```rust
structs.sort_by_key(|key| {
    return double(s.field);
});
```

This should not lint, since it works as expected: [Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d4a33a63a3feef6dbbd9e10bafc7f777)

Reply:
It does work, but it lints on "needless return"

Reply:
Please add a test for this, but add an ``allow`` for the needless return lint.

Reply:
Yes that's what I did :-) Waiting for rustup